### PR TITLE
Fix CompositionContract.Equals for null

### PR DESCRIPTION
**Pull request by hughbe**
- And remove workaround from fixed xunit bug
- And increase code coverage to 100%

### CODE REVIEW

File: src/System.Composition.Runtime/tests/CompositionContractTests.cs [hunk 1]
```diff
@@ -145,6 +145,44 @@ public static IEnumerable<object[]> Equals_TestData()
...
+            };
+
+            yield return new object[]
+            {
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", new object[0] } }),
+                new CompositionContract(typeof(int), "contractName", new Dictionary<string, object> { { "key", null } }),
+                false
+            };
+
+            if (!PlatformDetection.IsFullFramework)
```
Comment (hughbe) [inline src/System.Composition.Runtime/tests/CompositionContractTests.cs]: Is this necessary? Not sure if netfx uses the corefx code or not

File: src/System.Composition.Runtime/src/System/Composition/Hosting/Core/CompositionContract.cs [hunk 2]
```diff
@@ -186,8 +186,11 @@ internal static bool ConstraintEqual(IDictionary<string, object> first, IDiction
                 if (!second.TryGetValue(firstItem.Key, out object secondValue))
                     return false;
 
-                if (firstItem.Value == null && secondValue != null ||
-                    secondValue == null && firstItem.Value != null)
+                if (firstItem.Value == null)
+                {
+                    return secondValue == null;
+                }
+                else if (secondValue == null)
```
Comment (hughbe) [inline src/System.Composition.Runtime/src/System/Composition/Hosting/Core/CompositionContract.cs]: NRE was thrown below in line 205 as we call `firstItem.Value.Equals(secondValue)`

**Comment by karelz** [comment]
@maryamariyan @ViktorHofer can you please take a look? It's 3 weeks old PR ... (we didn't tag it, so you didn't know about it)

**Comment by karelz** [comment]
Thanks @maryamariyan, can you please merge it - given that CI is green?

File: src/System.Composition.Runtime/tests/CompositionContractTests.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/System.Composition.Runtime/tests/CompositionContractTests.cs]: S.C.Runtime s a netstandard assembly so it doesn't harm to test it here.

**Pull request merged**